<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="rviz"  default="false"/>
  <arg name="init_xarm" default="false" />

  <!-- roslaunch dashgo_moveit_config dashgo_xarm.launch robot_ip:=192.168.31.180 -->
  <include if="$(arg init_xarm)" file="$(find dashgo_moveit_config)/launch/dashgo_xarm.launch">
    <arg name="robot_ip" value="192.168.31.180" />
  </include>

  <!-- Load Params According to Robot -->
  <rosparam file="$(find pick_and_place)/config/params_DASHGO.yaml" command="load" />

  <!-- Pick & place server -->
  <!-- <node name="pick_and_place_server" pkg="pick_and_place" type="pick_and_place_server.py" output="screen" /> -->

  <!-- 2D Detector -->
  <!-- <include file="$(find object_detector)/launch/object_2D.launch">
    <arg name="SOURCE" value="/zed2/zed_node/left/image_rect_color" />
    <arg name="DEPTH_INPUT" value="/zed2/zed_node/depth/depth_registered" />
    <arg name="CAMERA_INFO" value="/zed2/zed_node/depth/camera_info" />
    <arg name="ROS_INPUT" value="True" />
    <arg name="USE_ACTIVE_FLAG" value="False" />
    <arg name="VERBOSE" value="False" />
    <arg name="DEPTH_ACTIVE" value="True" />
    <arg name="MODELS_PATH" default="$(find object_detector)/models/gazebo/" />
    <arg name="LABELS_PATH" default="$(find object_detector)/models/gazebo/label_map.pbtxt" />
    <arg name="CAMERA_FRAME" value="zed2_left_camera_optical_frame" />
    <arg name="USE_FAKE_DETECTOR" value="True" />
  </include> -->
  
  <!-- Follow Point -->
  <node name="follow_point" pkg="nav_main" type="FollowPoint.py" output="screen" respawn="true" args="" />
  
  <!-- MediaPipe - Human Pose -->
  <node name="poseEstimation" pkg="human_analyzer" type="poseEstimation.py" output="screen" respawn="true" args="" />

  <!-- 3D Detector -->
  <include file="$(find object_detector)/launch/object_3D_BAG.launch">
    <arg name="MAP_FRAME" default="map" />
    <arg name="CAMERA_FRAME" default="zed2_down_left_camera_frame" />
    <arg name="BASE_FRAME" default="base_link" />
    <arg name="map_to_odom"  default="true"/>
    <arg name="odom_to_base_footprint"  default="true"/>
    <arg name="POINT_CLOUD_TOPIC" default="/zed2_down/zed_down_node/point_cloud/ds_no_plane_cloud_registered" />
  </include>

  <!-- MainEngine -->
  <node name="CarryMyLuggage" pkg="main_engine" type="CarryMyLuggage.py" output="screen" />
  
  <!-- GPD -->
	<node name="detect_grasps_server_samples" pkg="gpd_ros" type="detect_grasps_server_samples" output="screen" respawn="true" args="">
    <param name="config_file" value="/catkin_home/src/manipulation/pick_and_place/config/dashgo_eigen_params.cfg" />
    <param name="rviz_topic" value="plot_grasps" />
  </node>
  

  <!-- Manipulation Client | Running in separate shell -->
  <node name="rviz" pkg="rviz" type="rviz" if="$(arg rviz)"
          args="-d $(find pick_and_place)/config/all.rviz"/>


</launch>
